pyomo/core/tests/unit/test_logical_expr.py

```diff
@@ -30,7 +30,7 @@
 class TestGenerate_RelationalExpression(unittest.TestCase):
 
     def setUp(self):
-        m = AbstractModel()
+        m = ConcreteModel()
```

Comment:
Why change from `Abstract` to `Concrete` here?

Reply:
It was necessary during development when I was close to getting matrix opreations between indexed Pyomo objects working.  However, that feature ended up being pulled from this PR due to other challenges.  I have reverted the change here.